packages/react/src/createConfig.ts

```diff
@@ -1,11 +1,73 @@
 import type { GlueStackConfig } from './types';
-
+import { convertStyledToStyledVerbosed } from './convertSxToSxVerbosed';
+import { stableHash } from './stableHash';
+import { propertyTokenMap } from './propertyTokenMap';
+import { updateOrderUnResolvedMap } from './updateOrderUnResolvedMap';
 export const createConfig = <
-  //@ts-ignore
-  T extends GlueStackConfig<T['tokens'], T['aliases'], T['globalStyle']>
+  T extends GlueStackConfig<
+    //@ts-ignore
+    T['tokens'],
+    T['aliases'],
+    T['globalStyle']
+  >
 >(
-  //@ts-ignore
-  config: T | GlueStackConfig<T['tokens'], T['aliases'], T['globalStyle']>
+  config:
+    | T
+    | GlueStackConfig<
+        //@ts-ignore
+        T['tokens'],
+        T['aliases'],
+        T['globalStyle']
+      >
 ): T => {
-  return config as any;
+  // @ts-ignore
+  if (!config.components) {
+    return config as any;
+  }
+  const newConfig = resolveComponentThemes(config);
+  return newConfig as any;
+};
+
+const resolveComponentThemes = (config: any) => {
+  const newConfig = { ...config };
+  delete config.components;
+
+  Object.keys(newConfig?.components ?? {}).forEach((componentName) => {
+    const component = newConfig.components[componentName];
+    if (component.theme) {
+      component.theme = resolveTheme(
+        component.theme,
+        {
```

Comment:
This can be moved to the top instead of creating it every time.

Reply:
Resolved with latest commit.